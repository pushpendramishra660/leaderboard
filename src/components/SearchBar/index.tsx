import React from 'react';
import styled from 'styled-components/native';
import {Image} from 'react-native';
import Constants from '../../constants';
import theme from '../../theme';

const SearchBarContainer = styled.View`
  background-color: ${theme.colors.antiFlashWhite};
  padding: 10px;
  margin: 10px;
  border-radius: 8px;
  flex-direction: row;
  align-items: center;
`;

const SearchIcon = styled(Image)`
  width: 20px;
  height: 20px;
  margin-right: 10px;
`;

const SearchInput = styled.TextInput`
  width: 60%;
  font-size: 16px;
`;

interface SearchBarProps {
  placeholder: string;
  onChangeText: (text: string) => void;
}

const SearchBar: React.FC<SearchBarProps> = ({placeholder, onChangeText}) => {
  return (
    <SearchBarContainer>
      <SearchIcon source={Constants.Images.searchIcon} resizeMode="contain" />
      <SearchInput
        placeholder={placeholder}
        placeholderTextColor={theme.colors.dark}
        onChangeText={onChangeText}
      />
    </SearchBarContainer>
  );
};

export default SearchBar;
