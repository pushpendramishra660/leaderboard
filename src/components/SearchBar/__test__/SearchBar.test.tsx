import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import SearchBar from '../index';

describe('<SearchBar />', () => {
  it('renders SearchBar correctly', () => {
    const {getByPlaceholderText} = render(
      <SearchBar placeholder="Search..." onChangeText={() => {}} />,
    );
    const inputElement = getByPlaceholderText('Search...');
    expect(inputElement).toBeTruthy();
  });
  it('calls the provided onChangeText function when input value changes', () => {
    const onChangeTextMock = jest.fn();
    const {getByPlaceholderText} = render(
      <SearchBar placeholder="Search..." onChangeText={onChangeTextMock} />,
    );
    const inputElement = getByPlaceholderText('Search...');
    fireEvent.changeText(inputElement, 'Test');
    expect(onChangeTextMock).toHaveBeenCalledWith('Test');
  });

  it('SearchBar component matches snapshot', () => {
    const onChangeTextMock = jest.fn();
    const component = renderer.create(
      <SearchBar placeholder="Search..." onChangeText={onChangeTextMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
