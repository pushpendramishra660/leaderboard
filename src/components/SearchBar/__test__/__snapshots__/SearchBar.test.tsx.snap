// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> SearchBar component matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F0F0F0",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../app/assets/images/search.png",
      }
    }
    style={
      Object {
        "height": 20,
        "marginRight": 10,
        "width": 20,
      }
    }
  />
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Search..."
    placeholderTextColor="#666D87BF"
    style={
      Object {
        "fontSize": 16,
        "width": "60%",
      }
    }
  />
</View>
`;
